d2game/d2player/key_binding_menu.go

```diff
@@ -0,0 +1,587 @@
+				}
+
+				primaryKeyLabelWrapper := settingLayout.AddLayout(d2gui.PositionTypeAbsolute)
+				primaryKeyLabelWrapper.SetSize(190, 0)
+				primaryLabel, _ := primaryKeyLabelWrapper.AddLabelWithColor(primaryStr, d2gui.FontStyleFormal11Units, primaryCol)
+				primaryLabel.SetHoverColor(d2util.Color(0x5450D1FF))
+
+				bl.primaryLabel = primaryLabel
+				bl.primaryLayout = primaryKeyLabelWrapper
+				bl.gameEvent = setting.gameEvent
+
+				secondaryKeyLabelWrapper := settingLayout.AddLayout(d2gui.PositionTypeAbsolute)
+				secondaryKeyLabelWrapper.SetSize(90, 0)
+				secondaryLabel, _ := secondaryKeyLabelWrapper.AddLabelWithColor(secondaryStr, d2gui.FontStyleFormal11Units, secondaryCol)
+				secondaryLabel.SetHoverColor(d2util.Color(0x5450D1FF))
+
+				bl.secondaryLabel = secondaryLabel
+				bl.secondaryLayout = secondaryKeyLabelWrapper
+				bl.binding = binding
+			}
+
+			menu.bindingLayouts = append(menu.bindingLayouts, &bl)
+		}
+
+		if i < len(groups)-1 {
+			layout.AddSpacerStatic(0, 25)
+		}
+	}
+
+	return wrapper
+}
+
+func (menu *KeyBindingMenu) getKeyColor(key d2enum.Key) color.RGBA {
+	switch key {
+	case -1:
+		return d2util.Color(0x555555FF)
+	default:
+		return d2util.Color(0xA1925DFF)
+	}
+}
+
+func (menu *KeyBindingMenu) OnMouseButtonDown(event d2interface.MouseEvent) {
+	if menu.Box.OnMouseButtonDown(event) {
+		return
+	}
+
+	if menu.scrollbar != nil && menu.scrollbar.IsInSliderRect(event.X(), event.Y()) {
+		menu.scrollbar.SetSliderClicked(true)
+		menu.scrollbar.onSliderMouseClick(event)
+		return
+	}
+
+	for _, bl := range menu.bindingLayouts {
+		gameEvent, typ := bl.GetPointedLayoutAndLabel(event.X(), event.Y())
+		if gameEvent != -1 {
+			if menu.currentBindingLayout != nil {
+				menu.lastBindingLayout = menu.currentBindingLayout
+				menu.currentBindingLayout.Reset()
+			}
+
+			menu.currentBindingModifier = gameEvent
+			menu.currentBindingLayout = bl
+			if typ == KeyBindingTypePrimary {
+				menu.currentBindingLayout.primaryLabel.SetIsBlinking(true)
+			} else if typ == KeyBindingTypeSecondary {
+				menu.currentBindingLayout.secondaryLabel.SetIsBlinking(true)
+			}
+			menu.currentBindingModifierType = typ
+			menu.isAwaitingKeyDown = true
+
+			bl.descLabel.SetIsHovered(true)
+			bl.primaryLabel.SetIsHovered(true)
+			bl.secondaryLabel.SetIsHovered(true)
+
+			return
+		} else if menu.currentBindingLayout != nil {
+			menu.currentBindingLayout.Reset()
+			menu.currentBindingLayout = nil
+			menu.isAwaitingKeyDown = false
+		}
+	}
+}
+
+func (menu *KeyBindingMenu) OnMouseMove(event d2interface.MouseMoveEvent) {
+	if menu.scrollbar != nil {
+		menu.scrollbar.onMouseMove(event)
+	}
+}
+
+func (menu *KeyBindingMenu) OnMouseButtonUp(event d2interface.MouseEvent) {
+	if menu.scrollbar != nil {
+		menu.scrollbar.SetSliderClicked(false)
+	}
+}
+
+func (menu *KeyBindingMenu) saveKeyChange(key d2enum.Key) {
+	var (
+		existingBinding *KeyBinding
+		gameEvent       d2enum.GameEvent
+		bindingType     KeyBindingType = -1
```

Comment:
I think that we should make `-1` a named constant to make its meaning more apparent